openwisp_monitoring/check/apps.py

```diff
@@ -1,8 +1,38 @@
 from django.apps import AppConfig
+from django.conf import settings
+from django.core.cache import cache
 from django.utils.translation import ugettext_lazy as _
+from django_netjsonconfig.signals import checksum_requested, config_download_requested
 
 
 class CheckConfig(AppConfig):
     name = 'openwisp_monitoring.check'
     label = 'check'
     verbose_name = _('Network Monitoring Checks')
+
+    def ready(self):
+        from openwisp_controller.config.models import Device
+        from ..device.models import DeviceData
+        from ..device.signals import device_metrics_received
+        checksum_requested.connect(
+            self.check_cache_key,
+            sender=Device,
+            dispatch_uid='checksum_requested'
+        )
+        config_download_requested.connect(
+            self.check_cache_key,
+            sender=Device,
+            dispatch_uid='config_download_requested'
+        )
+        device_metrics_received.connect(
+            self.check_cache_key,
+            sender=DeviceData,
+            dispatch_uid='received_device_metrics'
+        )
+
+    @classmethod
+    def check_cache_key(cls, instance, **kwargs):
+        from .tasks import find_related_checks
+        if getattr(settings, 'OPENWISP_MONITORING_SPONTANEOUS_RECOVERY_DETECTION', True):
```

Comment:
I'd use `OPENWISP_MONITORING_DEVICE_RECOVERY_DETECTION`, it makes more and more sense to move this to the device subapp.